core/src/main/java/org/apache/accumulo/core/metadata/TabletFile.java

```diff
@@ -104,21 +110,50 @@ public String getFileName() {
   }
 
   /**
-   * Exact string that is stored in the metadata table
+   * Return a string for reading the tablet file. Doesn't have to be exact string in metadata.
    */
-  public String getMetadataEntry() {
-    return metadataEntry;
+  public String getMetaRead() {
+    return normalizedPath;
   }
 
   /**
-   * Exact string that is stored in the metadata table but as a Text object
+   * Return a string for inserting a new tablet file.
    */
-  public Text getMetadataText() {
-    return new Text(metadataEntry);
+  public String getMetaInsert() {
+    return normalizedPath;
   }
 
-  public String getNormalizedPath() {
-    return normalizedPath;
+  /**
+   * Return a new Text object of {@link #getMetaInsert()}
+   */
+  public Text getMetaInsertText() {
+    return new Text(getMetaInsert());
+  }
+
+  /**
+   * New file was written to metadata so update the entry.
+   */
+  public void inserted() {
+    this.metadataEntry = Optional.of(normalizedPath);
```

Comment:
It would be nice to keep this object immutable.  Could possibly have this method return a new object instead of mutating itself, but I am not sure if that is easy to do.  If my suggestion is really cumbersome to implement, please ignore this comment.

Reply:
I will give it a try since I think it is worth exploring.

Reply:
@keith-turner I am not sure how returning a new mutable object will work with TabletMutatorBase [here](https://github.com/apache/accumulo/blob/def68d4593cfaac1bc4b82194440a2f80e6643d4/server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java#L76)...  any ideas?

Reply:
It looks like I will just have to find a good spot in the code to call ```insert()``` which will return the new StoredTabletFile object.  Thankfully a lot of the util methods we use for mutations return void so I was thinking to make them return collections of the new StoredTabletFiles, such as [here](https://github.com/apache/accumulo/blob/def68d4593cfaac1bc4b82194440a2f80e6643d4/server/base/src/main/java/org/apache/accumulo/server/util/MasterMetadataUtil.java#L65).